solr/core/src/java/org/apache/solr/schema/SchemaField.java

```diff
@@ -154,7 +154,33 @@ public SortField getSortField(boolean top) {
     return type.getSortField(this, top);
   }
 
-  /** 
+  /**
+   * @return The PostingsFormat name declared as parameter of this field; if it is not
+   * defined, the PostingsFormat name declared as parameter of this field type; or
+   * finally null if the parameter is not defined anywhere.
+   */
+  public String getPostingsFormat() {
+    String postingsFormat = (String) args.get(POSTINGS_FORMAT);
+    if (postingsFormat == null) {
+      postingsFormat = type.getPostingsFormat();
+    }
+    return postingsFormat;
+  }
+
+  /**
+   * @return The DocValuesFormat name declared as parameter of this field; if it is not
+   * defined, the DocValuesFormat name declared as parameter of this field type; or
+   * finally null if the parameter is not defined anywhere.
+   */
+  public String getDocValuesFormat() {
```

Comment:
Sure.
Note that to do that, I'll need to change the generic type of the "args" field. It must not be Map<String, ?> but Map<String, Object> to be allowed to use Map.getOrDefault(). I'll do that because the change is not exposed: the class is final and the args visibility is package.

Reply:
Interesting.  Sometimes Java generics rules elude me.  BTW even if args were exposed, Solr internals back-compat is a judgement call in terms of how common we think it is.